components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/context/util/OrganizationManagementUtil.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core.context.util;
+
+import org.apache.catalina.connector.Request;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.context.IdentityContext;
+import org.wso2.carbon.identity.core.context.model.Organization;
+import org.wso2.carbon.identity.core.context.model.RootOrganization;
+import org.wso2.carbon.identity.core.internal.IdentityCoreServiceDataHolder;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.organization.management.service.constant.OrganizationManagementConstants;
+import org.wso2.carbon.identity.organization.management.service.exception.OrganizationManagementException;
+import org.wso2.carbon.identity.organization.management.service.model.BasicOrganization;
+import org.wso2.carbon.user.api.Tenant;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for managing organization operations for the IdentityContextCreatorValve.
+ */
+public class OrganizationManagementUtil {
+
+    private static final Log LOG = LogFactory.getLog(OrganizationManagementUtil.class);
+    private static final String TENANT_SEPARATOR = "/t/";
+    private static final String ORG_SEPARATOR = "/o/";
+    private static final Pattern PATTERN_ORG_CONTEXT = Pattern.compile("^/o/[a-f0-9\\-]+?");
+    private static final Pattern PATTERN_ORG_CONTEXT_IN_TENANT_PERSPECTIVE =
```

Comment:
I would rather rename this as ORG_CONTEXT_IN_TENANT_CONTEXT or ORG_CONTEXT_WITHIIN_TENANT_CONTEXT